tests/raftstore/server.rs

```diff
@@ -167,7 +167,21 @@ impl Simulator for ServerCluster {
 
         conn.set_read_timeout(Some(timeout)).unwrap();
         let mut resp_msg = Message::new();
-        let get_msg_id = try!(rpc::decode_msg(&mut conn, &mut resp_msg));
+        let get_msg_id = try!(rpc::decode_msg(&mut conn, &mut resp_msg).map_err(|e| {
+            if let CodecError::Io(ref err) = e {
+                // If read timeout, the error will throw "Resource temporarily unavailable",
```

Comment:
> EAGAIN or EWOULDBLOCK
The socket is marked nonblocking and the receive operation would block, or a receive timeout had been set and the timeout expired before data was received.  POSIX.1 allows either error to be returned for this case, and does not require these constants to have the same value, so a portable application should check for both possibilities.

Quoted from man `recv`.